libsolidity/codegen/CompilerContext.h

```diff
@@ -278,7 +278,11 @@ class CompilerContext
 	void appendAuxiliaryData(bytes const& _data) { m_asm->appendAuxiliaryDataToEnd(_data); }
 
 	/// Run optimisation step.
-	void optimise(OptimiserSettings const& _settings) { m_asm->optimise(translateOptimiserSettings(_settings)); }
+	void optimise(OptimiserSettings const& _settings)
+	{
+		auto translatedSettings = evmasm::Assembly::OptimiserSettings::translate(_settings, m_evmVersion);
```

Comment:
Why introduce a local variable?

Reply:
I thought it was a little bit more readable that way. The call to get settings is long and hard to read and in my opinion makes it harder to see that it's just `optimise()` with some settings. But overall it's not such a big difference. I changed it to a version with no intermediate variable now.